compiler/src/iree/compiler/Codegen/Common/TransformDialectStrategies.cpp

```diff
@@ -6,44 +6,23 @@
 
 #include "iree/compiler/Codegen/Common/TransformDialectStrategies.h"
 
-#include <numeric>
-#include <type_traits>
-
 #include "iree-dialects/Dialect/LinalgTransform/StructuredTransformOpsExt.h"
 #include "iree-dialects/Transforms/TransformMatchers.h"
 #include "iree/compiler/Codegen/Common/TransformExtensions/CommonExtensions.h"
+// Needed because ApplyPatternOp is used everywhere.
+// TODO: Separate out GPU-specific patterns.
 #include "iree/compiler/Codegen/LLVMGPU/TransformExtensions/LLVMGPUExtensions.h"
-#include "iree/compiler/Codegen/PassDetail.h"
 #include "iree/compiler/Codegen/Passes.h"
-#include "iree/compiler/Dialect/Flow/IR/FlowDialect.h"
 #include "iree/compiler/Dialect/Flow/IR/FlowOps.h"
-#include "llvm/Support/CommandLine.h"
 #include "llvm/Support/Debug.h"
-#include "mlir/Analysis/SliceAnalysis.h"
-#include "mlir/Dialect/Affine/IR/AffineOps.h"
-#include "mlir/Dialect/Arith/IR/Arith.h"
-#include "mlir/Dialect/Bufferization/IR/Bufferization.h"
 #include "mlir/Dialect/Func/IR/FuncOps.h"
+// Needed because we only have gpu::GPUBlockMappingAttr available atm.
+// TODO: IREE should have its own attributes.
 #include "mlir/Dialect/GPU/IR/GPUDialect.h"
```

Comment:
Maybe we can move block distribution strategy to `TransformDialectStrategiesGPU.cpp`? This doesn't necessarily help because that file is still under `Codegen/Common`.

Reply:
You're right, addressed in a followup.

Re. moving out of `Codegen/Common`, I'd wait a bit until people get up to speed to avoid distributing tranforms around the code base too early.